<?xml version="1.0" encoding="ISO-8859-15"?>

<!-- Have a look at the documentation to see which resolution
and fps settings are supported:
https://click.intel.com/media/ZR300-Product-Datasheet-Public-002.pdf -->

<launch>
  <node pkg="maplab_realsense" type="zr300_node" name="zr300_node">

    <param name="imu/enabled" value="true"/>
    <param name="imu/imu_skip_first_n_gyro_measurements" value="100"/>

    <param name="fisheye/enabled" value="false"/>
    <param name="fisheye/width" value="640"/>
    <param name="fisheye/height" value="480"/>
    <param name="fisheye/fps" value="30"/>
    <param name="fisheye/enable_auto_exposure" value="true"/>
    <param name="fisheye/exposure_ms" value="25.0"/>
    <param name="fisheye/gain" value="9.0"/>
    <param name="fisheye/subsample_factor" value="1"/>

    <param name="color/enabled" value="true"/>
    <param name="color/width" value="640"/>
    <param name="color/height" value="480"/>
    <param name="color/fps" value="30"/>
    <param name="color/subsample_factor" value="1"/>

    <param name="depth/enabled" value="true"/>
    <param name="depth/width" value="640"/>
    <param name="depth/height" value="480"/>
    <param name="depth/fps" value="30"/>
    <param name="depth/subsample_factor" value="1"/>
    <param name="depth/median_filter_enabled" value="true"/>
    <!-- Only 3 or 5 are supported by opencv for this image type. -->
    <param name="depth/median_filter_size" value="5"/>
    <param name="depth/min_max_filter_enabled" value="true"/>
    <param name="depth/min_max_filter_size" value="3"/>
    <param name="depth/min_max_filter_threshold" value="0.1"/>

    <param name="infrared/enabled" value="true"/>
    <param name="infrared/subsample_factor" value="1"/>
    <param name="infrared/enable_auto_exposure" value="false"/>
    <param name="infrared/emitter_enabled" value="true"/>
    <!-- Values for outdoor calibration, disable autoexposure
         for these settings to take effect. -->
    <param name="infrared/gain" value="100"/>
    <param name="infrared/exposure" value="164"/>

    <param name="pointcloud/enabled" value="false"/>
  </node>

  <!-- Disable compression for depth -->
  <group ns="zr300_node/depth/image_raw" >
    <rosparam param="disable_pub_plugins">
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>
  </group>

</launch>
